### PR TITLE
[PoC]Basic listing of courses and assignments in the admin pages

### PR DESCRIPTION
**Pull request by marcospri**

### CODE REVIEW

File: lms/templates/admin/assignment/show.html.jinja2 [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{% import "lms:templates/admin/macros.html.jinja2" as macros %}
+{% extends "lms:templates/admin/base.html.jinja2" %}
+{% block header %}Assignment {{ assignment.id }}{% endblock %}
+{% block content %}
+<div>
+    <fieldset>
```
Comment (marcospri) [inline lms/templates/admin/assignment/show.html.jinja2:6]: Not showing much info here, just having a page to add on stuff.

File: lms/templates/admin/course/show.html.jinja2 [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{% block content %}
+<div>
+    <fieldset>
+        <legend class="label has-text-centered">Course</legend>
+            {{ macros.disabled_text_field("ID", course.id) }}
+            {{ macros.disabled_text_field("Name", course.name) }}
+            {{ macros.created_updated_fields(course) }}
+    </fieldset>
+    
+    <fieldset class="box">
```
Comment (marcospri) [inline lms/templates/admin/course/show.html.jinja2:13]: I don't think we need to search for assignments. In all LMS the assignments are all listed in the course page. We can do the same.

File: lms/templates/admin/organization/show.html.jinja2 [hunk 3]
```diff
@@ -77,6 +80,18 @@
                         </div>
                     </div>
                 </li>
+                <li class="tab-panel" data-section-id="course">
+                    <legend class="label has-text-centered">Find courses</legend>
+                    <form method="POST" action="{{ request.route_url("admin.courses") }}">
```
Comment (marcospri) [inline lms/templates/admin/organization/show.html.jinja2:85]: I should have handled posting this to the tab page.

File: lms/templates/admin/organization/show.html.jinja2 [hunk 4]
```diff
@@ -77,6 +80,18 @@
                         </div>
                     </div>
                 </li>
+                <li class="tab-panel" data-section-id="course">
```
Comment (marcospri) [inline lms/templates/admin/organization/show.html.jinja2:83]: This is on organization and not applications instance.

That seem to contradict how things are organized on the DB but the DB schema is wrong. We should have only one grouping per org that gets linked to different applications instances that have launched it.

Now we get one course row per applications instance that has ever seen it which complicates things for querying /reporting.


If we also need to query them at the AI level we could use a couple of macros and have it in both places.

File: lms/views/admin/course.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        course_id = self.request.matchdict["id_"]
+        course = self._get_course_or_404(course_id)
+
+        return {
+            "course": course,
+            "assignments": self.assignment_service.get_course_assignments(course),
+        }
+
+    """
+    @view_config(
```
Comment (marcospri) [inline lms/views/admin/course.py:45]: Copy pasted from Orgs.

File: lms/services/course.py [hunk 6]
```diff
@@ -191,6 +219,8 @@ def _get_copied_from_course(self, lti_params) -> Course | None:
 def course_service_factory(_context, request):
     return CourseService(
         db=request.db,
-        application_instance=request.lti_user.application_instance,
+        application_instance=(
```
[the remark this answers is not in the document]
Comment (acelaya) [reply lms/services/course.py:222]: Yep

**Pull request closed without merging**